psycop/common/feature_generation/data_checks/flattened/feature_describer_tsflattener_v2.py

```diff
@@ -0,0 +1,89 @@
+import re
+from dataclasses import dataclass
+from typing import Any, Callable
+
+import polars as pl
+
+
+@dataclass
+class ParsedPredictorColumn:
+    col_name: str
+    feature_name: str
+    fallback: str
+    time_interval_start: str
+    time_interval_end: str
+    time_interval_format: str
+    resolve_multiple_strategy: str
+    is_static: bool
+
+
+def tsflattener_v2_column_is_static(col_name: str) -> bool:
+    str_match = "within"
+    return str_match not in col_name
+
+
+def parse_predictor_column_name(
+    col_name: str,
+    is_static: Callable[[str], bool] = tsflattener_v2_column_is_static,
+    feature_name_regex: str = r"[a-zA-Z]+_(.*?)_(?=fallback|within)",
+    fallback_regex: str = r"_fallback_(.+)$",
+    time_interval_start_regex: str = r"_within_([0-9]+)",
+    time_interval_end_regex: str = "_to_([0-9]+)_",
+    time_interval_format_regex: str = r"to_[0-9]+_([a-z]+)_",
+    resolve_multiple_strategy_regex: str = r"([a-z]+)_fallback",
```

Comment:
Haha, you sure went all-in on dependency injection/parametrisation here! ❤️ 

Makes the code super-reusable, even if we end up changing the format again ❤️